src/bot.test.ts

```diff
@@ -26,41 +26,43 @@ describe("Bot", () => {
         initializeTelegramFn: sinon.spy(),
         initializeEmailFn: sinon.spy(),
         initializeSpacesFn: sinon.spy(),
-        shouldContinueFn: () => {
-          const returned = !isShouldContinueCalled;
-          isShouldContinueCalled = true;
-          return returned;
-        },
+        initializeGracefulShutdownFn: sinon.spy(),
         processSpacesFn: sinon.spy(),
-        waitForFn: () => Promise.resolve(),
+        waitForFn: waitFor,
+        batchCooldown: 300,
       };
     });
 
+    afterEach(() => {
+      // Trigger a shutdown event so the function stops processing blocks
+      depsMock.emitter.emit(BotEventNames.GRACEFUL_SHUTDOWN_START);
+    });
+
     it("should emit the expected events", async () => {
-      const detectEvent = (name: BotEventNames) =>
```

Comment:
I'm removing this snippet as it is similar to an existing one, `resolveOnEvent`, that can reused